src/parser/parser.go

```diff
@@ -0,0 +1,1228 @@
+	}
+	return &nodes.Compare{
+		Expr:       expr,
+		Ops:        ops,
+		NodeCommon: nodes.NodeCommon{Lineno: lineno},
+	}, nil
+}
+
+func (p *parser) parseMath1() (nodes.Node, error) {
+	lineno := p.stream.Current().Lineno
+	left, err := p.parseConcat()
+	if err != nil {
+		return nil, err
+	}
+
+	for p.stream.Current().Type == lexer.TokenAdd || p.stream.Current().Type == lexer.TokenSub {
+		currentType := p.stream.Current().Type
+		p.stream.Next()
+		right, err := p.parseConcat()
+		if err != nil {
+			return nil, err
+		}
+		left = makeMathNode(left, right, currentType, lineno)
+		lineno = p.stream.Current().Lineno
+	}
+	return left, nil
+}
+
+func (p *parser) parseMath2() (nodes.Node, error) {
+	lineno := p.stream.Current().Lineno
+	left, err := p.parsePow()
+	if err != nil {
+		return nil, err
+	}
+
+	for p.stream.Current().Type == lexer.TokenMul || p.stream.Current().Type == lexer.TokenDiv || p.stream.Current().Type == lexer.TokenFloordiv || p.stream.Current().Type == lexer.TokenMod {
+		currentType := p.stream.Current().Type
+		p.stream.Next()
+		right, err := p.parsePow()
+		if err != nil {
+			return nil, err
+		}
+		left = makeMathNode(left, right, currentType, lineno)
+		lineno = p.stream.Current().Lineno
+	}
+	return left, nil
+
+}
+
+func (p *parser) parseConcat() (nodes.Node, error) {
+	lineno := p.stream.Current().Lineno
+	left, err := p.parseMath2()
+	if err != nil {
+		return nil, err
+	}
+	args := []nodes.Node{left}
+
+	for p.stream.Current().Type == lexer.TokenTilde {
+		p.stream.Next()
+		right, err := p.parseMath2()
+		if err != nil {
+			return nil, err
+		}
+		args = append(args, right)
+	}
+	if len(args) == 1 {
+		return args[0], nil
+	}
+	return &nodes.Concat{
+		Nodes:      args,
+		NodeCommon: nodes.NodeCommon{Lineno: lineno},
+	}, nil
+}
+
+func (p *parser) parsePow() (nodes.Node, error) {
+	lineno := p.stream.Current().Lineno
+	left, err := p.parseUnary(true)
+	if err != nil {
+		return nil, err
+	}
+
+	for p.stream.Current().Type == lexer.TokenPow {
+		p.stream.Next()
+		right, err := p.parseUnary(true)
+		if err != nil {
+			return nil, err
+		}
+		left = &nodes.Pow{
+			Left:       left,
+			Right:      right,
+			NodeCommon: nodes.NodeCommon{Lineno: lineno},
+		}
+		lineno = p.stream.Current().Lineno
+	}
+	return left, nil
+}
+
+func (p *parser) parseUnary(withFilter bool) (nodes.Node, error) {
+	//         token_type = self.stream.current.type
+	//        lineno = self.stream.current.lineno
+	//        node: nodes.Expr
+	//
+	//        if token_type == "sub":
+	//            next(self.stream)
+	//            node = nodes.Neg(self.parse_unary(False), lineno=lineno)
+	//        elif token_type == "add":
+	//            next(self.stream)
+	//            node = nodes.Pos(self.parse_unary(False), lineno=lineno)
+	//        else:
+	//            node = self.parse_primary()
+	//        node = self.parse_postfix(node)
+	//        if with_filter:
+	//            node = self.parse_filter_expr(node)
+	//        return node
```

Comment:
this comment should be removed

Reply:
I wonder how it made it here, I must have collapsed it in the IDE...